zproject/backends.py

```diff
@@ -1003,6 +1009,31 @@ def get_verified_emails(self, *args: Any, **kwargs: Any) -> List[str]:
             verified_emails.append(details["email"])
         return verified_emails
 
+class SAMLAuthBackend(SocialAuthMixin, SAMLAuth):
+    auth_backend_name = "SAML"
+
+    def auth_complete(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
+        """
+        Additional ugly wrapping on top of auth_complete in SocialAuthMixin.
+        If the /complete/saml endpoint is accessed, but RelayState isn't specified
+        in POST, django.utils.datastructures.MultiValueDictKeyError will be raised
+        inside python-social-auth saml module, which doesn't convert it into something
+        more reasonable.
+        If it is accessed with RelayState specified, but missing SAMLResponse,
+        OneLogin_Saml2_Error ends up raised.
+        We need to catch and handle those here.
+        """
+        try:
+            # Call the auth_complete method of SocialAuthMixIn
+            return super().auth_complete(*args, **kwargs)  # type: ignore # monkey-patching
+        except MultiValueDictKeyError as e:
+            logging.info("SAML authentication: MultiValueDictKeyError: " + str(e))
+            return None
+        except OneLogin_Saml2_Error as e:
+            # This error provides a proper description of the issue, so we can simply log str(e)
+            logging.info(str(e))
```

Comment:
The meaning of the errors above will tell us how these should be logged.  If it's likely a bug in some software, we probably want at least `logging.warning`.  If it's something that would routinely happen without a bug (e.g. someone hitting the `Back` button at the wrong time that recovers automatically), then `logging.debug` or `.info` might be most appropriate.

Reply:
As explained in the extra paragraph added to the comment, these are caused by stuff like the user going directly to ``/complete/saml/``, so I think ``.info`` is right here?